src/controllers/trainer.js

```diff
@@ -45,43 +62,47 @@ const updateTrainer = (req, res) => {
     firstName, lastName, dni, phone, email, city, password, salary, isActive,
   } = req.body;
 
-  return Trainer.findByIdAndUpdate(
-    id,
-    {
-      firstName,
-      lastName,
-      dni,
-      phone,
-      email,
-      city,
-      password,
-      salary,
-      isActive,
-    },
-    { new: true },
-  )
-    .then((trainer) => {
-      if (!trainer) {
-        return res.status(404).json({
-          message: `Trainer with the id: ${id} was not fount`,
-          data: undefined,
-          error: true,
-        });
-      }
-      return res.status(201).json({
-        message: 'Trainer updated succesfully',
-        data: trainer,
-        error: false,
+  try {
+    const actualTrainer = await Trainer.findById(id);
+
+    if (!actualTrainer) {
+      return res.status(404).json({
+        message: `The trainer with the ID: ${id} doesn't exists`,
+        data: undefined,
+        error: true,
       });
-    })
-    .catch((error) => res.status(400).json({
-      message: 'An error ocurred',
+    }
+
```

Comment:
Could you check if the email or dni entered already exists? Like u did on the post controller

Reply:
Great addition! But this validation is still missing. I meant something like:
```
  const found = await Trainer.findOne({ $or: [{ dni }, { email }] });

    if (found) {
      return res.status(400).json({
        message: `The admin with DNI ${dni} or Email ${email} already exists`,
        data: undefined,
        error: true,
      });
    }
```